### PR TITLE
Fix #2321: Add new objects to OpenShift model (with operators.coreos.com and monitoring.coreos.com APIs)

### LINKED ISSUE #2321: Can I create a Catalog Source and Subscription using the Kubernetes or the OpenShift Client
Hi ,
This could be more of a question rather than an issue .However I am not able to figure out how to make it work.I would like to install a CatalogSource and Subscription to the cluster using the client,

Below is the code snippet I am using - 

```
void createResourcefromYaml(File resourceYaml){
KubernetesClient client = new DefaultKubernetesClient()
FileInputStream fileInputStream = new FileInputStream(resourceYaml)
client.load(fileInputStream).createOrReplace()
}

```
Below is sample catalogsource.yaml

```
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: some-name
spec:
  sourceType: valid source type
  image: some image
  imagePullPolicy: Always
  displayName: some name
  publisher: some name

```

However I get the below error 
```
Caused by: com.fasterxml.jackson.databind.JsonMappingException: No resource type found for:operators.coreos.com/v1alpha1#CatalogSource
 at [Source: (BufferedInputStream); line: 11, column: 1]
	at com.fasterxml.jackson.databind.JsonMappingException.from(JsonMappingException.java:274)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.fromObjectNode(KubernetesDeserializer.java:85)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:57)
	at io.fabric8.kubernetes.internal.KubernetesDeserializer.deserialize(KubernetesDeserializer.java:46)
	at com.fasterxml.jackson.databind.ObjectReader._bindAndClose(ObjectReader.java:1719)
	at com.fasterxml.jackson.databind.ObjectReader.readValue(ObjectReader.java:1228)
	at io.fabric8.kubernetes.client.utils.Serialization.unmarshal(Serialization.java:138)
	... 7 more
```

I can do the above without any issue through commandline 
`oc apply -f catalog-source.yaml`
but not via the client.

Could you advice what am I missing here please?

Thank you

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fixes #2321

+ Add new OpenShift resources(whichever applicable and not colliding with existing resources) in OpenShift Model as per listed
  resources in https://docs.openshift.com/container-platform/4.4/rest_api/operator_apis/operator-apis-index.html

+ Added these new modules in kubernetes-model-generator:
  - openshift-console-model
  - openshift-monitoring-model
  - openshift-operatorhub-model
  - openshift-operator-model
+ Added new endpoints in openshift-client DSL:
  - client.config() - config.openshift.io/v1
  - client.operator() - operator.openshift.io/v1 and operator.openshift.io/v1alpha1
  - client.operatorHub() - operators.coreos.com/v1alpha1
  - client.console() - console.openshift.io/v1
  - client.quota() - quota.openshift.io/v1
  - client.monitoring() - monitoring.coreosc.com/v1

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [X] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [X] I Added [CHANGELOG](../CHANGELOG.md) entry regarding this change
 - [X] I have implemented unit tests to cover my changes
 - [X] I have added/updated the [javadocs](https://www.javadoc.io/doc/io.fabric8/kubernetes-client/latest/index.html) and other [documentation](https://github.com/fabric8io/kubernetes-client/blob/master/doc/CHEATSHEET.md) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=fabric8io_kubernetes-client) report
 - [ ] I tested my code in Kubernetes
 - [X] I tested my code in OpenShift

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: kubernetes-model-generator/pom.xml [hunk 1]
```diff
@@ -53,6 +53,10 @@
     <module>kubernetes-model-settings</module>
     <module>kubernetes-model-storageclass</module>
     <module>openshift-model</module>
+    <module>openshift-operator-model</module>
+    <module>openshift-operatorhub-model</module>
+    <module>openshift-monitoring-model</module>
+    <module>openshift-console-model</module>
```
Comment (manusa) [inline kubernetes-model-generator/pom.xml:59]: nit: naming of modules/directories/etc. is not consistent with the other naming patterns

>>>> `openshift-model-*`

**Review by manusa — APPROVED** [review]

**Comment by manusa** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Comment by rohanKanojia** [comment]
[merge]

**Comment by manusa** [comment]
[merge]

**Pull request merged**